### PR TITLE
✨ Add presets country lists to amp-geo

### PR DESCRIPTION
**Pull request by jpettitt**
Adds preset country lists to `amp-geo`

### CODE REVIEW

**Comment by jpettitt** [comment]
@erwinmombay when the server side of this is ready please merge

File: extensions/amp-geo/0.1/amp-geo-presets.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    'YT', // Mayotte
+    'NC', // New Caledonia
+    'RE', // Reunion
+    'BL', // St Barthelemy
+    'MF', // St Martin
+    'PM', // St Pierre & Miquelon
+    'SJ', // Svalbard & Jan Mayen Islands
+    'VA', // Vatican City
+    'WF', // Wallis & Futuna Islands
+    'EZ', // Eurozone'
```
Comment (erwinmombay) [inline extensions/amp-geo/0.1/amp-geo-presets.js]: remove trailing` '`

File: extensions/amp-geo/0.1/amp-geo.js [hunk 2]
```diff
@@ -181,15 +191,36 @@ export class AmpGeo extends AMP.BaseElement {
...
   }
 
+  /**
+   * checkGroup_() does this.country_  match the group
+   * @param {!Array<string>} countryGroup The group to match against
+   * @return {boolean}
+   */
+  checkGroup_(countryGroup) {
+    let presetCountries = [];
+    countryGroup = countryGroup.map((country, idx) => {
```
Comment (jridgewell) [inline extensions/amp-geo/0.1/amp-geo.js]: This should have been a reduce, not a map.

File: extensions/amp-geo/0.1/amp-geo.js [hunk 3]
```diff
@@ -181,15 +191,36 @@ export class AmpGeo extends AMP.BaseElement {
...
+      if (/^preset-/.test(country)) {
+        user().assert(
+            isArray(ampGeoPresets[country]),
+            `<${TAG}> preset ${country} not found`);
+        delete countryGroup[idx];
```
Comment (jridgewell) [inline extensions/amp-geo/0.1/amp-geo.js]: This doesn't work the way you expect it.

File: extensions/amp-geo/0.1/amp-geo.js [hunk 4]
```diff
@@ -181,15 +191,36 @@ export class AmpGeo extends AMP.BaseElement {
...
+        presetCountries = presetCountries.concat(ampGeoPresets[country]);
+        return;
```
Comment (jridgewell) [inline extensions/amp-geo/0.1/amp-geo.js]: This inserts an `undefined` into the output.

Comment (jpettitt) [reply extensions/amp-geo/0.1/amp-geo.js]: Fixed in #16451

**Pull request merged**